lib/Controller/ImageController.php

```diff
@@ -0,0 +1,246 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2021 Julien Veyssier <eneiluj@posteo.net>
+ *
+ * @author Julien Veyssier <eneiluj@posteo.net>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Text\Controller;
+
+use Exception;
+use OCP\AppFramework\Http;
+use OCA\Text\Service\ImageService;
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Http\DataDisplayResponse;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\IRequest;
+use Psr\Log\LoggerInterface;
+
+class ImageController extends Controller {
+	public const IMAGE_MIME_TYPES = [
+		'image/png',
+		'image/jpeg',
+		'image/jpg',
+		'image/gif',
+		'image/x-xbitmap',
+		'image/x-ms-bmp',
+		'image/bmp',
+		'image/svg+xml',
+		'image/webp',
+	];
+
+	/**
+	 * @var string|null
+	 */
+	private $userId;
+	/**
+	 * @var ImageService
+	 */
+	private $imageService;
+	/**
+	 * @var LoggerInterface
+	 */
+	private $logger;
+
+	public function __construct(string $appName,
+								IRequest $request,
+								LoggerInterface $logger,
+								ImageService $imageService,
+								?string $userId) {
+		parent::__construct($appName, $request);
+		$this->userId = $userId;
+		$this->imageService = $imageService;
+		$this->request = $request;
+		$this->logger = $logger;
+	}
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * @param int $textFileId
+	 * @param string $imagePath
+	 * @return DataResponse
+	 */
+	public function insertImageFile(int $textFileId, string $imagePath): DataResponse {
```

Comment:
Thinking about that it might be useful to just make this a public endpoint for both logged in users and public share links and use the session token to authenticate. That way the same methods can also be used when opening a file through direct editing (ref https://github.com/nextcloud/server/pull/17625) through the desktop/Android/iOS clients.

FYI @tobiasKaminsky For inserting local files/files from an URL this should then work without any client change. The existing feature to insert an image from Nextcloud would though depend on https://github.com/nextcloud/text/issues/1170

Reply:
@juliushaertl Yes sure, seems better. Do you have any example where authentication is checked like that?